lib/Haskell/Extra/Erase.agda

```diff
@@ -14,8 +14,8 @@ module Haskell.Extra.Erase where
     @0 xs  : List a
 
   record Erase (@0 a : Set ℓ) : Set ℓ where
-    constructor Erased
-    field @0 get : a
+    instance constructor Erased
+    field @0 {{get}} : a
```

Comment:
What's the motivation for changing the interface of `Erase` and making the field an instance? Seems like it forces explicit brackets `{{}}` everywhere.

Reply:
Well, it was to make sure that instance search could actually find an element of type `Erase A` when there is an erased instance of type `A` in scope. But I found a better way to get the same effect, which does not break backwards compatibility (see newly pushed version).

Reply:
Is it a situation that arises often? Having instances of `a` laying around? For equality proofs I get, but otherwise not so sure. But I trust you that the interface hasn't changed :). (I would have kept the old `Erase` definition and defined `iErased` separately but it probably doesn't matter much. And with yours one can decide to match and directly introduce an instance, which I don't know if you can do otherwise)

Reply:
The opposite way (defining `Erased` as the constructor and `iErased` as the pattern synonym) is not accepted by Agda:
```
.../agda2hs/lib/Haskell/Extra/Erase.agda:22,21-22: error: [IllegalInstanceVariableInPatternSynonym]
Variable is bound as instance in pattern synonym, but does not
resolve as instance in pattern: x
when scope checking the declaration
  pattern iErased ⦃ x ⦄ = Erased x
  ```